### DISCUSSION TITLE
`xml-editor`: Change Save protocol for editing Formatters and Aggregators

### DISCUSSION THREAD

**Discussion by bronwyncombs** [specify/specify7#5893] [Feature Requests]
**Describe the problem**

When editing/adding formatters through app resources, the dialog button options are `Delete` and `Close`. The only option to save is back on the page to view available formats for the selected table. 
<img width="936" alt="Screenshot 2023-11-27 at 11 26 19 AM" src="https://github.com/specify/specify7/assets/135047322/b04679f2-067c-4cc7-bccb-d7de534e5070">

Saving or canceling edits per formatter on the table would be very helpful for users to keep track of their changes and know what's being saved/added. Having the cancel is also important so a user can confidently undo potentially mistaken changes/additions. 

**Describe the solution you'd like**

I would like to add save/cancel buttons when adding/editing formatter and aggregator. 

Editing would be easier with something more like the security and accounts UI

https://github.com/specify/specify7/assets/135047322/fa9fb030-7915-4beb-8890-c366481fc3db

**_(Except closing should return the user to the list of formatters for the table like it does currently with the dialog in xml-editor)_**

The Delete, Clone, Add, and Save buttons that appear at the bottom when browsing the table list should be removed. 
<img width="936" alt="Screenshot 2023-11-27 at 11 24 00 AM" src="https://github.com/specify/specify7/assets/135047322/73bd05db-0a5b-4645-9421-81d55069c4af">

Same for the buttons when viewing the list of formatters for a table once selected
<img width="936" alt="Screenshot 2023-11-27 at 11 25 07 AM" src="https://github.com/specify/specify7/assets/135047322/6551936e-c02c-4b0a-b9d9-7ff518e96afd">

**Comment by grantfitzsimmons** [comment]
Here's my take: I think we should move the delete button to the meta menu instead of showing it in the editor panel

<img width="750" alt="image" src="https://github.com/specify/specify7/assets/37256050/36f8b341-bef3-4e29-91ee-29acaa2287a5">

**Comment by maxpatiiuk** [comment]
I like the removing delete button idea.
Don't know how often that button is useful in practice, where as avoiding confusion and reducing accidental clicks would be beneficial.
And once "Close" is the only button in the dialog, it becomes clear that that's the button you have to press.
Clean and simple - I like it

Adding explicit "fake save", and cancel, alongside delete would work too, but is more cluttered and more complicated
